### PR TITLE
Add Mbed Crypto tests and test plan

### PR DESCRIPTION
**Pull request by jjohnson-arm**

### CODE REVIEW

File: ci/lava/tests/test-mbed-crypto.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Script to run systemd tests on the DUT."""
+
+import pytest
+import re
+import subprocess
```
Comment (petedyerarm) [inline ci/lava/tests/test-mbed-crypto.py]:
```suggestion

```

**Comment by petedyerarm** [comment]
https://lava.mbedcloudtesting.com/scheduler/job/66137 result using this branch

**Comment by jjohnson-arm** [comment]
Related to ARMmbed/mbl-tools#328

File: ci/lava/tests/test-mbed-crypto.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
```
Comment (diegorusso) [inline ci/lava/tests/test-mbed-crypto.py]: I guess in this file few things need to be changed. There are references to systemd which clearly don't have anything to do with it.

File: ci/lava/tests/mbed-crypto.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    parameters:
+        #
+        # virtual_env: specifies the Python virtual environment
+        #
+        virtual_env:
+
+run:
+    steps:
+        - . $virtual_env/bin/activate
+
```
Comment (diegorusso) [inline ci/lava/tests/mbed-crypto.yaml]:
```suggestion
```

File: ci/lava/tests/test-mbed-crypto.py (hunk 2 above)
Comment (jjohnson-arm) [reply ci/lava/tests/test-mbed-crypto.py]: Agreed!

**Comment by diegorusso** [comment]
> There are a couple of files which I don't understand:
> 
>     * ci/lava/tests/mbed-crypto-poky.sh
> 
>     * ci/lava/tests/mbed-crypto-poky.yaml
> 
> 
> Do we need those?

Pete just told me why. For poky we cannot run it with mbl-cli. Can we have a comment about that?

File: ci/lava/tests/test-mbed-crypto.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Script to run tests on the DUT."""
+
+import pytest
+import re
+
```
Comment (jjohnson-arm) [inline ci/lava/tests/test-mbed-crypto.py]:
```suggestion


```

**Comment by jjohnson-arm** [comment]
New commit message:
```
Add Mbed Crypto tests and test plan for MBL & poky

Mbed Crypto tests added to replace psa arch tests that have been disabled.
* Add support for MBL via pytest
* Add support for poky via shell script (as pytest not available)
```

**Comment by jjohnson-arm** [comment]
For MBL
* build: http://jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-warrior-dev/detail/jij-warrior-dev/84/pipeline/
* results: https://lava.mbedcloudtesting.com/scheduler/job/66599

For poky:
* build: http://jenkins.mbed-linux.arm.com/blue/organizations/jenkins/poky-psa-test/detail/poky-psa-test/93/pipeline/
* results: https://lava.mbedcloudtesting.com/scheduler/job/66600

**Comment by jjohnson-arm** [comment]
Building is fixed (of poky), testing needed a few tweaks but now works: https://lava.mbedcloudtesting.com/scheduler/job/66993

**Pull request merged**